tests/Unit/BasicSyntaxTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+use Mockery as m;
+use RicorocksDigitalAgency\Soap\Request\Request;
+use RicorocksDigitalAgency\Soap\Response\Response;
+
+it('can obtain a WSDL', function () {
+    $mock = m::mock(Request::class);
+    $mock->shouldReceive('beforeRequesting', 'afterRequesting', 'to')->andReturnSelf()
+        ->shouldReceive('functions')->andReturn(
+            [
+                'AddResponse Add(Add $parameters)',
+                'SubtractResponse Subtract(Subtract $parameters)',
+                'MultiplyResponse Multiply(Multiply $parameters)',
+                'DivideResponse Divide(Divide $parameters)',
+                'AddResponse Add(Add $parameters)',
+                'SubtractResponse Subtract(Subtract $parameters)',
+                'MultiplyResponse Multiply(Multiply $parameters)',
+                'DivideResponse Divide(Divide $parameters)',
+            ]
+        );
+
+    expect(soap(null, $mock)->to(EXAMPLE_SOAP_ENDPOINT)->functions())
+        ->toBeArray()
+        ->not->toBeEmpty();
+});
+
+it('can call a SOAP function')
+    ->fake(['*' => Response::new(['AddResult' => 35])])
+    ->expect(fn () => $this->soap()->to(EXAMPLE_SOAP_ENDPOINT)->Add(['intA' => 10, 'intB' => 25])->AddResult)
```

Comment:
I think for this test this should be `call('Add', ...)`

Reply:
Updated